File: backend/src/main/java/com/messmate/backend/dto/response/BalanceResponse.java
package com.messmate.backend.dto.response;

import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class BalanceResponse {
    private String userId;
    private String name;

    private Double totalExpenseShare;
    private Double totalPaidForBazar;

    private Double paymentsMade;
    private Double paymentsReceived;

    private Double netBalance; // > 0 means receive, < 0 means owe

    private Double pendingPaymentsMade;

    private String balanceMessage;
}

File: android/app/src/main/java/com/messmate/android/data/balance/BalanceResponse.kt
package com.messmate.android.data.balance

data class BalanceResponse(
    val userId: String,
    val name: String,
    val totalExpenseShare: Double?,
    val totalPaidForBazar: Double?,
    val paymentsMade: Double?,
    val paymentsReceived: Double?,
    val netBalance: Double?,
    val balanceMessage: String?
)
